### PR TITLE
Builder disk usage and clean

### PR DESCRIPTION
**Pull request by giggsoff**
We use dedicated docker container as builder and we are able to clean data inside only by re-creating of it. Let's add disk usage and clean commands for builder.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Review by deitch — DISMISSED** [review]

**Comment by deitch** [comment]
Which builder is this for? There is logic to use multiple builders based on architecture, with fallback for cross-compiling.

**Comment by giggsoff** [comment]
> Which builder is this for? There is logic to use multiple builders based on architecture, with fallback for cross-compiling.

Good point. I put additional flags in novel commands to be compatible with existing `pkg build` logic of builder select.

**Comment by deitch** [comment]
Two questions about the latest version.

First, should this go under `lkt pkg builder` and not `lkt builder`? The builders are used exclusively for building packages, which is entirely under `lkt pkg`. I am not quite set on it, as I can see how builder management might be its own sort of management thing (although, again, exclusively used for `lkt pkg build`), and we do stuff like cache management at the level of `lkt cache`. Then again, cache is the "bridge" between pkg and general build. Like to hear your thoughts on this.

Second, what is the flow for this, and therefore what are the arguments used for?

When I do a build, e.g. `lkt pkg build --platforms=linux/arm64,linux/amd64 foo/`, I am telling it which platforms to build for. It then uses the logic of finding the appropriate builders, based on the logic listed [here](https://github.com/linuxkit/linuxkit/blob/master/docs/packages.md#build-platforms).

What flow are you targeting? I am using `du`, but same logic for `prune` (and all future `builder` commands).

* `lkt builder du` - should give for all platforms (or just current one)?
* `lkt builder du --platforms=linux/arm64,linux/amd64` - should give for whatever builders currently are in use for arm64 and amd64 (which might be the same ones)?

I think your concern is that the last command won't necessarily map to what someone uses to build, because they might build with `lkt pkg build --platforms=linux/arm64,linux/amd64 --builders linux/arm64=some-special-context`, and so running the above du command would miss it. Is that it? Hence, you added in the `--builders` option to `du`, so you can do:

```
lkt builder du --builders ...
```

If I did understand it correctly, some things I am not clear on:

* why do we need `--builder-image`? We add it only because when launching an image, we need to allow the user to override the calculated defaults. But here, we are just reporting on what is there. I think this confuses the user. And if the image doesn't match, will you start a new one? I don't need that, as by default the new image has a usage of 0.
* `--arch` doesn't align with the `--platforms` used everywhere else.

File: src/cmd/linuxkit/builder.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package main
```
Comment (deitch) [inline src/cmd/linuxkit/builder.go:1]: This file should be named `pkg_builder.go`, in line with all of the other files that are subcommands of `pkg`.

File: src/cmd/linuxkit/builder.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	// Please keep these in alphabetical order
+	fmt.Printf("  du\n")
+	fmt.Printf("  prune\n")
+	fmt.Printf("\n")
+	fmt.Printf("'options' are the backend specific options.\n")
+	fmt.Printf("See '%s builder [command] --help' for details.\n\n", invoked)
+}
+
+// Process the builder
+func builder(args []string) {
```
Comment (deitch) [inline src/cmd/linuxkit/builder.go:28]: this should be named `pkgBuilder`, in line with the other `pkg` subcommand functions.

Probably also should be `pkgBuilderCommands()` and `pkgBuilderUsage()`.

File: src/cmd/linuxkit/pkglib/docker.go [hunk 3]
```diff
@@ -214,7 +215,7 @@ func (dr *dockerRunnerImpl) contextSupportCheck() error {
 // 1. if dockerContext is provided, try to create a builder with that context; if it succeeds, we are done; if not, return an error.
 // 2. try to find an existing named runner with the pattern; if it succeeds, we are done; if not, try next.
 // 3. try to create a generic builder using the default context named "linuxkit".
-func (dr *dockerRunnerImpl) builder(ctx context.Context, dockerContext, builderImage, platform string, restart bool) (*buildkitClient.Client, error) {
+func (dr *dockerRunnerImpl) builder(ctx context.Context, dockerContext, builderImage, platform string, restart bool) (interface{}, error) {
```
Comment (deitch) [inline src/cmd/linuxkit/pkglib/docker.go:218]: Why is this changed to return an `interface{}`? That is about as generic as it gets. Is there any reason it cannot return a `*buildkitClient.Client`?

**Review by deitch — CHANGES_REQUESTED** [review]

File: src/cmd/linuxkit/builder.go (hunk 1 above)
Comment (giggsoff) [reply src/cmd/linuxkit/builder.go]: Done

File: src/cmd/linuxkit/builder.go (hunk 2 above)
Comment (giggsoff) [reply src/cmd/linuxkit/builder.go]: Done, thank you

File: src/cmd/linuxkit/pkglib/docker.go (hunk 3 above)
Comment (giggsoff) [reply src/cmd/linuxkit/pkglib/docker.go]: Well, you are right, we have only one builder now, let's think about abstractions later.

**Review by deitch — APPROVED** [review]

**Comment by deitch** [comment]
Once CI is green, we can merge it in.
